src/Umbraco.Web/Extensions/UdiExtensions.cs

```diff
@@ -36,8 +36,30 @@ public static IPublishedContent ToPublishedContent(this Udi udi)
                         return umbracoHelper.TypedMember(memberAttempt.Result);
                     break;
             }
-            
+
             return null;
         }
+
+        /// <summary>
+        /// An extension method to easily acquire the integer Id for a given udi
+        /// </summary>
+        /// <param name="udi"></param>
+        /// <returns>An <see cref="int"/> identifier if the item is found, -1 otherwise</returns>
+        public static int ToIntId(this Udi udi)
```

Comment:
I appreciate that and I did look into `IEntityService` but I fail to see how that would help with people's querying needs. 
We'll end up with things like:

`var url = library.NiceUrl(ApplicationContext.Current.Services.EntityService.GetIdForUdi(myUdi).Id);`

It's long, impossible to discover how to do this (like it is now) plus we always tell people not to use `Services` directly in their templates. There has to be a better way.